saleor/graphql/notifications/tests/test_external_notification_query.py

```diff
@@ -0,0 +1,508 @@
+    )
+    assert_no_permission(response)
+    assert response.status_code == 200
+    sendgrid_plugin_notify.assert_not_called()
+    user_email_plugin_notify.assert_not_called()
+    admin_email_plugin_notify.assert_not_called()
+    webhook_plugin_notify.assert_not_called()
+
+
+@patch("saleor.plugins.sendgrid.plugin.SendgridEmailPlugin.notify")
+def test_notify_via_external_notification_trigger_with_extra_payload(
+    sendgrid_plugin_notify,
+    product_with_single_variant,
+    sendgrid_email_plugin,
+    external_notification_trigger_query,
+    staff_api_client,
+    permission_manage_products,
+):
+
+    test_json = {"TEST": "VALUE", "TEST_LIST": ["GUEST1", "GUEST2"]}
+    plugin = sendgrid_email_plugin(
+        active=True, api_key="AB12", account_password_reset_template_id="123"
+    )
+    variables = {
+        "input": {
+            "ids": [
+                to_global_id(ProductVariant.__name__, pk)
+                for pk in product_with_single_variant.variants.values_list(
+                    "pk", flat=True
+                )
+            ],
+            "extraPayloads": json.dumps(test_json),
+            "externalEventType": UserNotifyEvent.ORDER_CANCELED,
+        },
+        "pluginId": plugin.PLUGIN_ID,
+    }
+
+    response = staff_api_client.post_graphql(
+        external_notification_trigger_query,
+        variables,
+        permissions=[permission_manage_products],
+    )
+
+    assert response.status_code == 200
+    payload = sendgrid_plugin_notify.call_args[1]["payload"]
+    assert "TEST" in payload.keys()
+    assert "TEST_LIST" in payload.keys()
+    assert ["GUEST1", "GUEST2"] == payload.get("TEST_LIST")
+
+
+@patch("saleor.plugins.sendgrid.plugin.SendgridEmailPlugin.notify")
+def test_notify_via_external_notification_trigger_with_extra_payload_for_customers(
+    sendgrid_plugin_notify,
+    staff_users,
+    sendgrid_email_plugin,
+    external_notification_trigger_query,
+    staff_api_client,
+    permission_manage_users,
+):
+
+    test_json = {"TEST": "VALUE", "TEST_LIST": ["GUEST1", "GUEST2"]}
+    plugin = sendgrid_email_plugin(
+        active=True, api_key="AB12", account_password_reset_template_id="123"
+    )
+    variables = {
+        "input": {
+            "ids": [to_global_id(User.__name__, user.id) for user in staff_users],
+            "extraPayloads": json.dumps(test_json),
+            "externalEventType": UserNotifyEvent.ORDER_CANCELED,
+        },
+        "pluginId": plugin.PLUGIN_ID,
+    }
+
+    response = staff_api_client.post_graphql(
+        external_notification_trigger_query,
+        variables,
+        permissions=[permission_manage_users],
+    )
+
+    assert response.status_code == 200
+    payload = sendgrid_plugin_notify.call_args[1]["payload"]
+    assert "TEST" in payload.keys()
+    assert "TEST_LIST" in payload.keys()
+    assert ["GUEST1", "GUEST2"] == payload.get("TEST_LIST")
+
+
+class TestSetWithVariousDataTypes:
```

Comment:
We don't use TestCases. If you have something which is similar for preparing data for many tests then you can create a fixture for this.
I see that you have common part where you have an asserts. Not sure of this. I prefer to have explicit assert in each test, to see all conditions that the test needs to pass. 
If you really need this, then you can wrap it in a simple function. No need to create a class for this.